src/solvers/flattening/boolbv_case.cpp

```diff
@@ -49,15 +49,11 @@ bvt boolbvt::convert_case(const exprt &expr)
       break;
 
     case COMPARE:
-      if(compare_bv.size()!=op.size())
-      {
-        std::cerr << "compare operand: " << compare_bv.size()
-                  << "\noperand: " << op.size() << '\n'
-                  << it->pretty()
-                  << '\n';
-
-        throw "size of compare operand does not match";
-      }
+      DATA_INVARIANT(
```

Comment:
I'm still not sure I agree with this.  CHECK_RETURN (as it's the return value of a previous call to convert) or INVARIANT (as a generic operational property of the algorithm).